File: Competitive Programming/codeforces/training/B_Playing_with_GCD.py
# Copyright (c) 2023, Le Duc Phuc Long

# If you don't think twice, you have to code twice.

# Import session
import sys
input = sys.stdin.readline
from collections import defaultdict
from math import *

############ ---- Input Functions ---- ############
def inp():
    return int(input())

def inlt():
    return list(map(int, input().split()))

def instr():
    return list(input().strip())

def invr():
    return map(int, input().split())

############ ---- Other Functions ---- ############
# Precompute

# IO
#sys.stdin = open('in.txt', 'r')
#sys.stdout = open('out.txt', 'w')
# Main function
for _ in range(inp()):
    n = inp()
    a = inlt()
    a.insert(0, 1)
    a.append(1)
    
    b = [0]*(n+2)
    for i in range(1, n+2):
        b[i] = lcm(a[i], a[i-1])
        
    chk = True
    for i in range(1, n+1):
        if gcd(b[i], b[i+1]) != a[i]:
            chk = False
    if chk:
        print("YES")
    else:
        print("NO")
            

File: Competitive Programming/codeforces/training/B_Playing_with_GCD.cpp
// Copyright (c) 2023, Le Duc Phuc Long

/*If you don't think twice, you have to code twice.*/

// Include
#include <bits/stdc++.h>  // NOLINT
// #include <conio.h>

// Namespace
using namespace std;  // NOLINT

// Define
#define nln '\n'

// Typedef and Struct
typedef int64_t lo;

// Constant

// Declare global variables.
vector<pair<lo, lo>> dir4 = {{0, 1}, {0, -1}, {1, 0}, {-1, 0}};
// End of global variable declaration.

void precompute() {
}

lo __lcm(lo a, lo b) {
    return a*b/__gcd(a, b);
}

void solve() {
    lo n;   cin >>n;
    vector<lo> a(n+2, 1);
    for (lo i = 1; i <= n; ++i)
        cin >> a[i];

    vector<lo> b(n+2, 1);
    for (lo i = 1; i <= n+1; ++i)
        b[i] = __lcm(a[i], a[i-1]);
    
    bool chk = true;
    for (lo i = 1; i <= n; ++i)
        if (__gcd(b[i], b[i+1]) != a[i])
            chk = false;
    
    if (chk)
        cout << "YES" << nln;
    else
        cout << "NO" << nln;
}

int main(int argc, char* argv[]) {
    //freopen("in.txt", "r", stdin);
    //freopen("out.txt", "w", stdout);
    cin.tie(0)->sync_with_stdio(0);
    cout.tie(0)->sync_with_stdio(0);
    int T = 1;
    cin >> T;
    precompute();
    while (T--) {
        solve();
    }
    cerr << "It's ok Long, Good for now !!" << nln;
    return 0;
}